ext/sysvshm/sysvshm.stub.php

```diff
@@ -2,36 +2,22 @@
 
 /** @generate-function-entries */
 
-/** @return resource|false */
-function shm_attach(int $key, int $memsize = UNKNOWN, int $perm = 0666) {}
-
-/**
- * @param resource $shm_identifier
- */
-function shm_detach($shm_identifier): bool {}
-
-/**
- * @param resource $id
- */
-function shm_has_var($id, int $variable_key): bool {}
-
-/**
- * @param resource $shm_identifier
- */
-function shm_remove($shm_identifier): bool {}
-
-/**
- * @param resource $shm_identifier
- */
-function shm_put_var($shm_identifier, int $variable_key, $variable): bool {}
-
-/**
- * @param resource $id
- * @return mixed
- */
-function shm_get_var($id, int $variable_key) {}
-
-/**
- * @param resource $id
- */
-function shm_remove_var($id, int $variable_key): bool {}
+final class Sysvshm
+{
+}
+
+/** @return Sysvshm|false */
+function shm_attach(int $key, int $memsize = UNKNOWN, int $perm = 0666): Sysvshm|false {}
+
+function shm_detach(Sysvshm $shm_identifier): bool {}
```

Comment:
Maybe rename `$shm_identifier` to just `$shm`?

Reply:
Yes, I initially thought about dealing with it later (during mass changes), but it definitely makes sense now (I've already pushed a commit). Also, is the `Sysvshm` class name makes sense? Maybe if we added an OO API, a more meaningful name would make more sense.

Reply:
A more descriptive name is always good IMHO, the closest I can personally guess what Sysvshm means is "System Virtual Shared Memory" and that took some thinking

Reply:
I'll have to admit that I didn't think about the class name at all. For the recent `Sysvsem` PR it would probably make sense to call it `SysvSemaphore`. I think for this one `Sysvshm` is okay, but better spelled as `SysvSHM`. `SysvSharedMemory` would also be okay, if you want to be more explicit.

Reply:
For me, `SysvSemaphore` certainly seems good.

Until recently, when I started contributing to PHP, I didn't really know what `SHM` means, that's why I'd lean towards `SysvSharedMemory`, since it's much more explicit a name for web programmers. :)